source/extensions/filters/udp/udp_proxy/config.h

```diff
@@ -151,8 +158,12 @@ class UdpProxyFilterConfigImpl : public UdpProxyFilterConfig,
 
   // UdpSessionFilterChainFactory
   void createFilterChain(Network::UdpSessionFilterChainFactoryCallbacks& callbacks) const override {
-    for (const Network::UdpSessionFilterFactoryCb& factory : filter_factories_) {
-      factory(callbacks);
+    for (const auto& filter_config_provider : filter_factories_) {
+      auto config = filter_config_provider->config();
+      if (config.has_value()) {
```

Comment:
Should there be some warning/error for a missing config here?

Reply:
agree, added a log

Reply:
Thanks. Can you update the docs in the UDP listener, saying that the missing config in ECDS results in a no-op warning? For other filters, it is protocol-specific, e.g. 503 for HTTP.

Reply:
ECDS isn't really getting supported in this PR - the API remains the same, requiring consumers to configure the filter statically. In next PR I'll add dynamic consumption that uses discovery, and I'll update docs accordingly.
Also note that this kind of change applies to UDP **session** filters and not regular UDP listener filters